odoo/addons/base/models/res_users.py

```diff
@@ -162,7 +162,8 @@ def _compute_full_name(self):
     def _search_full_name(self, operator, operand):
         lst = True
         if isinstance(operand, bool):
-            domains = [[('name', operator, operand)], [('category_id.name', operator, operand)]]
+            category_ids = self.env['ir.module.category'].sudo().search([('name', operator, operand)]).ids
+            domains = [[('name', operator, operand)], [('category_id', 'in', category_ids)]]
```

Comment:
I may be mistaken but I'm not convinced keeping that one makes sense: this is an `isinstance(operand, bool)`, so the operand can be either `True` or `False`. The former makes no sense, so so this should be either `!= False` (aka is set) or `= False` (aka is not set). 

ModuleCategory's `name` is `required`, so these are respectively `True` and `False`, always.

The next conditional is `True` if `!= False`, therefore the resulting domain is either

    AND(name != False, True)

or

    OR(name = False, False)

In both cases, the term on the category seems useless.